File: app/src/main/java/com/code_design_camp/client/friday/HeadDisplayClient/dialog/ProgressDialog.java
package com.code_design_camp.client.friday.HeadDisplayClient.dialog;

import android.app.Activity;
import android.content.Context;
import android.view.View;
import android.widget.TextView;

import com.code_design_camp.client.friday.HeadDisplayClient.R;

import androidx.annotation.StringRes;
import androidx.appcompat.app.AlertDialog;

public class ProgressDialog extends AlertDialog {
    String msg;
    Context context;
    View v;
    TextView message_text;
    public ProgressDialog(Context context, String message) {
        super(context);
        this.msg = message;
        this.context = context;
        v = ((Activity) context).getLayoutInflater().inflate(R.layout.loading_dialog, null);
        message_text = v.findViewById(R.id.msg_text);
        message_text.setText(message);
        super.setView(v);
        super.setCancelable(false);
    }

    public void setMessage(String message) {
        this.msg = message;
        message_text.setText(message);
    }

    public void setMessage(@StringRes int stringResource) {
        this.msg = getContext().getString(stringResource);
        message_text.setText(msg);
    }
}

File: appModules/core-ui/src/main/java/com/friday/ar/core_ui/dialog/ProgressDialog.kt
package com.friday.ar.core_ui.dialog

import android.app.Activity
import android.view.View
import androidx.annotation.StringRes
import androidx.appcompat.app.AlertDialog
import com.friday.ar.core_ui.R
import kotlinx.android.synthetic.main.loading_dialog.view.*

/**
 * A custom ProgressDialog.
 * Androids internal [android.app.ProgressDialog] is deprecated due to the change of the material design guidelines.
 *
 * @see android.app.ProgressDialog
 *
 * @see [Material guidelines](material.io)
 */
class ProgressDialog(internal var context: Activity, private var msg: String) : AlertDialog(context) {
    internal var v: View = (context).layoutInflater.inflate(R.layout.loading_dialog, null)

    init {
        super.setView(v)
        super.setCancelable(false)
        v.msg_text.text = msg
    }

    fun setMessage(message: String) {
        this.msg = message
        v.msg_text.text = message
    }

    fun setMessage(@StringRes stringResource: Int) {
        this.msg = getContext().getString(stringResource)
        v.msg_text.text = msg
    }
}
